sectorstatemgr/sectorstatemgr.go

```diff
@@ -214,13 +231,99 @@ func (m *SectorStateMgr) refreshState(ctx context.Context) (*SectorStateUpdates,
 	}
 	activeSectors := make(map[abi.SectorID]struct{}, len(activeSet))
 	for _, info := range activeSet {
-		sectorID := abi.SectorID{
+		sid := abi.SectorID{
+			Miner:  abi.ActorID(mid),
+			Number: info.SectorNumber,
+		}
+
+		activeSectors[sid] = struct{}{}
+	}
+
+	committedSectorsSet, err := m.fullnodeApi.StateMinerSectors(ctx, m.Maddr, nil, head.Key())
+	if err != nil {
+		return nil, err
+	}
+	committedSectors := make(map[abi.SectorID]struct{}, len(committedSectorsSet))
+	for _, info := range committedSectorsSet {
+		sid := abi.SectorID{
 			Miner:  abi.ActorID(mid),
 			Number: info.SectorNumber,
 		}
 
-		activeSectors[sectorID] = struct{}{}
+		committedSectors[sid] = struct{}{}
+	}
+
+	sectorNumbers, err := m.minerApi.SectorsList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Slice(sectorNumbers, func(i, j int) bool {
+		return sectorNumbers[i] < sectorNumbers[j]
+	})
+
+	sectorsTable := make([]Sector, 0, len(sectorNumbers))
+	for _, sectorNumber := range sectorNumbers {
+		sid := abi.SectorID{
+			Miner:  abi.ActorID(mid),
+			Number: sectorNumber,
+		}
+
+		//st, err := m.minerApi.SectorsStatus(ctx, sectorNumber, true)
```

Comment:
@dirkmc I think we should limit ourselves to calling only APIs for all sectors that we have high confidence will respond within up to 30sec.

If we want to individually call `SectorsStatus` for all sectors, it should happen in a background worker/subsystem, and we need to cache the state and introduce concurrency.

`2M sectors * 10ms per response / 8 concurrency factor` is still 40min.